src/js/player.js

```diff
@@ -1900,6 +1900,18 @@ class Player extends Component {
   }
 
   /**
+   * Reset the player. Removes all sources, loads the Html tech, and calls `load`.
+   *
+   * @return {Player} Returns the player
+   * @method reset
+   */
+  reset() {
+    this.loadTech_('Html5', null);
```

Comment:
I was debating that but
1. we would need to implement something for flash in this case and probably would need to modify the swf
2. Setting a source should choose the appropriate tech after the reset
3. Html5 is basically the default tech.

Switching to Html5 and then resetting it is probably the easiest way to get to a nice base state where no sources are loaded and all readyStates are reset.